src/Servers/IIS/AspNetCoreModuleV2/CommonLib/application.h

```diff
@@ -57,14 +57,16 @@ class APPLICATION : public IAPPLICATION
     VOID
     Stop(bool fServerInitiated) override
     {
-        SRWExclusiveLock stopLock(m_stateLock);
-
-        if (m_fStopCalled)
         {
-            return;
-        }
+            SRWExclusiveLock stopLock(m_stateLock);
```

Comment:
There was a reason we help lock for the entire duration of StopInternal... I wish I remembered why.

Reply:
#YOLOSWAGGINZ

Reply:
It was used for out of process. I did the minimal and re-added the lock around StopInternal().

Reply:
Now Stop retakes the lock twice for out of process. What was it used for there?

Reply:
Ahh I now see why we locked the entire call to StopAsync. It makes it so it is guaranteed that the app can be replaced by the time the request returns 503. 

By having the lock in Stop, it means that the next call to TryCreateHandler will not finish until the app is shutdown, meaning the dotnet process has shut down by the time TryCreateHandler returns. 

By removing the lock, we don't affect any behavior, we simply will server AppOffline earlier. However, files may still be locked for a bit.

Reply:
Won't it brake web deploy scenarios?

Reply:
I don't think so, but we can discuss tomorrow.

Reply:
We figured out why. It's to prevent a second shutdown request from immediately returning and crashing the process. I'll _try_ to write a test but it's probably going to be racy.